includes/Generator/Order.php

```diff
@@ -236,17 +243,26 @@ public static function batch( $amount, array $args = array() ) {
 			return $amount;
 		}
 
+		// Pre-generate dates if date-start is provided
+		// This ensures chronological order: lower order IDs = earlier dates
+		if ( ! empty( $args['date-start'] ) ) {
+			self::$batch_dates = self::generate_batch_dates( $amount, $args );
+		}
+
 		$order_ids = array();
 
 		for ( $i = 1; $i <= $amount; $i ++ ) {
 			$order = self::generate( true, $args );
-			if ( ! $order ) {
+			if ( ! $order instanceof \WC_Order ) {
 				error_log( "Batch generation failed: Order {$i} of {$amount} could not be generated" );
 				continue;
 			}
 			$order_ids[] = $order->get_id();
 		}
 
+		// Clear batch dates after generation
+		self::$batch_dates = null;
```

Comment:
Instead of using a static variable to store the batch dates, what do you think about just passing the batch dates as an argument to `generate` method? This way, we don't need to worry about shared state between different instances of the class. We can add a new parameter, or modify `$assoc_args` to include the batch dates.